File: src/main/java/com/cuupa/classificator/services/kb/Token.java
package com.cuupa.classificator.services.kb;

import java.util.List;

public abstract class Token {
	
	protected List<String> tokenValue;

	public abstract boolean match(String text);

	public void setTokenValue(List<String> tokenValue) {
		this.tokenValue = tokenValue;
	}
}

File: src/main/java/com/cuupa/classificator/knowledgebase/services/token/Token.kt
package com.cuupa.classificator.knowledgebase.services.token

abstract class Token {

    var tokenValue= mutableListOf<String>()

    abstract fun match(text: String?): Boolean

    abstract val distance: Int

    abstract fun clone(): Token

    override fun toString(): String {
        return tokenValue.joinToString(",", "", "")
    }
}